src/parser/util.js

```diff
@@ -65,7 +65,18 @@ pp.isLineBreak = function () {
 pp.canInsertSemicolon = function () {
   return this.match(tt.eof) ||
     this.match(tt.braceR) ||
-    this.isLineBreak();
+    this.isLineBreak() ||
+    (this.hasPlugin("lightscript") && (
+      // LSC oneline statement ASI cases
+      // Allow if x: throw y else: throw z
+      this.match(tt._else) ||
+      this.match(tt._elif) ||
+      // Allow (-> throw new Error)()
+      this.match(tt.parenR) ||
+      // Technically it is legal to insert a ; after a ;.
+      // Allows -> throw new Error; f()
+      this.state.tokens[this.state.tokens.length - 1].type === tt.semi
```

Comment:
The `;` after `;` thing is the result of going down a pretty deep rabbit hole. In JS, `(x) => y; z` is legal (compiles to a `Body` of two `ExpressionStatement`s) and I wanted to maintain consistency.

So that would mean `(x) => throw new Error; z` should compile to a `Body` of two `ExpressionStatements` as well. (In Coffeescript, that actually compiles to one function with two statements in the body)

Ultimately the thing that was preventing that was the absence of this line of code here, so I added it, and it had the result I wanted without breaking JS, so I went with it. Not sure about better options.